.github/workflows/test.yml

```diff
@@ -70,11 +63,42 @@ jobs:
       - uses: actions/setup-go@v3.0.0
         with:
           go-version: 1.17
-
       - uses: actions/checkout@v2
-
+      - uses: technote-space/get-diff-action@v6.0.1
+        with:
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
       - name: Build Docker Image
         run: make docker-build-debug
-
       - name: Test E2E
         run: make test-e2e
+
+  liveness-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+      - uses: technote-space/get-diff-action@v6.0.1
+        with:
+          PATTERNS: |
+            **/**.go
+            go.mod
+            go.sum
+      - name: Install Ignite CLI
+        run: |
+          curl https://get.ignite.com/cli@v0.20.0! | bash
+        if: env.GIT_DIFF
+      - name: Start Local Network via Ignite CLI
+        run: |
+          make start-localnet-ci > ignite.out 2>&1 &
+        if: env.GIT_DIFF
+      - name: Test Local Network Liveness
+        run: |
+          sleep 3m
```

Comment:
why sleep ? and sleep so long ?

Reply:
`ignite` start a blocking process where it runs a chain, so we run the process in the background and pipe both stdout and stderr to a file. `ignite` takes a bit to startup (builds app, builds proto, etc...), so the 3m is just a safe "buffer" to wait for `ignite` to actually start the chain.